cmapPy/math/modz.py

```diff
@@ -0,0 +1,68 @@
+'''
+modz.py
+
+Given a matrix of profiles on which you want to calculate a weighted average, returns a single profile of modz values
+'''
+
+import pandas as pd
+import numpy as np
+import os
+import math
+
+def upper_triangle(correlation_matrix):
+    '''
+    :param correlation_matrix (pandas df): Correlations between all replicates
+    :return upper_tri_series (pandas series): Upper triangle extracted from corr mat
+    '''
+    upper_triangle = correlation_matrix.where(np.triu(np.ones(correlation_matrix.shape), k=1).astype(np.bool))
+
+    # convert matrix into long form description
+    upper_tri_series = upper_triangle.stack().reset_index(level=1)
+
+    upper_tri_series.columns = ['rid', 'spearman_corr']
+
+    # Index at this point is CID, it now becomes a column
+    upper_tri_series.reset_index(level=0, inplace=True)
+
+    return upper_tri_series.round(4)
```

Comment:
Why round?

Reply:
Ahhh this is something I've had requested for things in merino for a long time. Usually I just leave it to be however many decimal places it gives me but others have been requesting 4 dp to make it more manageable when looking at the data

Reply:
Maybe rounding could be an option?